test/lib/graphql/schema_comparator/diff/field_test.rb

```diff
@@ -0,0 +1,175 @@
+require "test_helper"
+
+describe GraphQL::SchemaComparator::Diff::Field do
+  let(:type) do
+    GraphQL::ObjectType.define do
+      name "Foo"
+    end
+  end
+
+  let(:differ) { GraphQL::SchemaComparator::Diff::Field.new(type, type, old_field, new_field) }
+  let(:changes) { differ.diff }
+  let(:change) { differ.diff.first }
+
+  describe "#diff" do
+    describe "field type change" do
+      let(:old_field) do
+        GraphQL::Field.define do
+          name "foo"
+          type types.String
+        end
+      end
+
+      let(:new_field) do
+        old_field.redefine { type types.Boolean }
+      end
+
+      it "is a breaking change" do
+        assert_equal 1, changes.size
+        assert change.breaking?
+
+        assert_equal "Field `Foo.foo` changed type from `String` to `Boolean`", change.message
+      end
+    end
+
+    describe "field type change from scalar to list of the same type" do
+      let(:old_field) do
+        GraphQL::Field.define do
+          name "bar"
+          type types[types.String]
+        end
+      end
+
+      let(:new_field) do
+        old_field.redefine { type types.String }
+      end
+
+      it "is a non-breaking change" do
+        assert_equal 1, changes.size
+        assert change.breaking?
+
+        assert_equal "Field `Foo.bar` changed type from `[String]` to `String`", change.message
+      end
+    end
+
+    describe "field type change from non-null to null of the same underyling type" do
```

Comment:
Is this a change we would consider `Dangerous` vs `Breaking`?

Reply:
the JS implementation still has this as a breaking change: https://github.com/graphql/graphql-js/blob/fed3ef58fde674fa96d79879eb4dfc24a0ba1db0/src/utilities/__tests__/findBreakingChanges-test.js#L232-L234